stonesoup/types/state.py

```diff
@@ -156,18 +156,22 @@ class ParticleState(Type):
     This is a particle state object which describes the state as a
     distribution of particles"""
 
-    timestamp = Property(datetime.datetime, default=None,
-                         doc="Timestamp of the state. Default None.")
     particles = Property([Particle],
                          doc='List of particles representing state')
+    timestamp = Property(datetime.datetime, default=None,
+                         doc="Timestamp of the state. Default None.")
+
+    @property
+    def ndim(self):
+        return self.particles[0].ndim
 
     @property
     def mean(self):
```

Comment:
Agree it would be good to do this. I've always worried that caching could be difficult with mutable list, with mutable particles with mutable state vector. A lot of places for it to go wrong.

I think @jmbarr's proposal in #160 to use just an array for particles (e.g. `StateVectors` in #204) would make these easier, as you could change the [arrays `writable` flag](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.flags.html).